src/index.js

```diff
@@ -1,27 +1,42 @@
 // @flow
 
 // Helpers
-export function createToken(name: string): any {
-  // $FlowFixMe
-  return () => {
-    throw new Error(`Missing required value for token: ${name}.`);
-  };
+export const TokenType = {
+  Required: 0,
+  Optional: 1,
+};
+function Ref() {}
+class TokenImpl {
+  name: string;
+  ref: mixed;
+  type: $Values<typeof TokenType>;
+
+  constructor(name: string, ref: mixed) {
+    this.name = name;
+    this.ref = ref || new Ref();
+    this.type = ref ? TokenType.Optional : TokenType.Required;
+    if (!ref) {
+      // $FlowFixMe
+      this.optional = new TokenImpl(name, this.ref);
+    }
+  }
 }
 
-export function createOptionalToken<Default>(
-  name: string,
-  defaultValue: Default
-): Default {
+export type Token<T> = {
+  (): T,
+  optional: () => ?T,
+};
+export function createToken(name: string): Token<any> {
   // $FlowFixMe
```

Comment:
Same, do we still need FloxFixMe here?

Reply:
This one we'll still need.  Flow (correctly) asserts that the shape of `TokenImpl` doesn't match that of the return type here (`Token<any>`).  With the exception of `fusion-core/base-app`, we want consumers to assume that all tokens ascribe to the `Token` type.  It takes a little bit of typing magic here to make it all work, even though Flow is technically correct that these types don't match.